[owen-yap] iP

owen-yap opened the pull request:
# Dukey
> “Java is to Javascript what car is to Carpet.” – Chris Heilmann

- George Washington
* John Adams
+ Thomas Jefferson

Do this:
1. download from [here](https://github.com/owen-yap/ip/releases)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

Features code like this:
```
public static void main(String[] args) {
        new Duke("data.txt").run();
    }
```

and even more spectacular `code` like this:
```
public void mark() {
      this.isDone = true;
}
```
For a low price of ~~$929192921201~~ **FREE**, you can:
- [x] add task
- [ ] complete tasks
- [ ] task task

Commit:
Add Gradle support

owen-yap committed:
level-1 greet, echo, exit

owen-yap committed:
level-2: Add, List

owen-yap committed:
level-3 mark as done

owen-yap committed:
level-4 todo, deadline, event

owen-yap committed:
added automated i/o text ui tests

owen-yap committed:
implemented minimal error handling

owen-yap committed:
implemented delete feature

owen-yap committed:
implemented serialization of data into text file

owen-yap committed:
implement date time

owen-yap committed:
fix merge conflict with branch-Level-8

owen-yap committed:
Refactor code to use more OOP

owen-yap committed:
Refactor code under duke package

owen-yap committed:
Merge remote-tracking branch 'origin/add-gradle-support'

owen-yap committed:
Add gradle support and checkstyle

owen-yap committed:
Add Junit Tests for Todo and Deadline

owen-yap committed:
Modify gradle file for packaging

joytqt-1202 commented on src/main/java/Duke.java:
```diff
@@ -130,15 +156,19 @@ public static void main(String[] args) {
                             break;
                         case "deadline":
```
Could you remove the indentation for each of the switch case statements?
i.e.
```
switch(cmd[0]) {
case "deadline": 
    ...
case "event":
    ...
```

joytqt-1202 commented on src/main/java/Deadline.java:
```diff
@@ -1,14 +1,16 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
 public class Deadline extends Task {
```
Should there be java docs for this class?

owen-yap replied:
Done!

joytqt-1202 commented on src/main/java/Duke.java:
```diff
@@ -1,6 +1,9 @@
 import exceptions.DukeException;
 
 import java.io.*;
```
Perhaps avoid using wildcard imports

owen-yap replied:
Done!

joytqt-1202 commented on src/main/java/Event.java:
```diff
@@ -1,16 +1,20 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
 public class Event extends Task {
```
This public class could use a java doc too

owen-yap replied:
Done!

joytqt-1202 commented on src/main/java/Event.java:
```diff
@@ -1,16 +1,20 @@
         this.to = to;
     }
 
     @Override
```
Could make use of inherit doc to inherit the overridden public method's java docs
i.e.
```
/**
 * {@inheritDoc}
 * /
```

joytqt-1202 reviewed (commented):
Just a few comments on java coding conventions

Lanreath commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,127 @@
+
+            TaskList data = this.storage.loadTasks();
+
+            switch (cmd) {
```
Maybe the input handlers can be abstracted as different classes with separate methods for easier future extension?

owen-yap committed:
Add Javadoc header comments to all methods

Lanreath commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,63 @@
+
+    public void saveTasks(TaskList tasks) {
+        try {
+            OutputStream output = new FileOutputStream("./data.txt");
```
I enjoyed the use of serialisation to store and retrieve the tasks without the need for manual input and output 👍

owen-yap replied:
thanks

Lanreath commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,42 @@
+        sc = new Scanner(System.in);
+    }
+    public void printMsg(String[] msgs) {
+        System.out.println("____________________________________________________________");
```
Could you reduce the repetition of dividers? 🤔

owen-yap committed:
Make stylistic changes

Lanreath commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,127 @@
+        try {
+            date = LocalDateTime.parse(dateString, formatter);
+        } catch (DateTimeParseException e) {
+            throw new IllegalArgumentException("Invalid date format: " + dateString + ". Please give your date" +
```
Throwing this exception does not help to identify the specific input error. Is there any way to prevent this?

owen-yap replied:
Correct date format will be displayed to the user so that they know that the date format of the input is wrong.

Lanreath commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,29 @@
+
```
I noticed multiple instances of incorrect import formatting.

owen-yap replied:
Done!

Lanreath reviewed (commented):
Well done, though more can be done to simplify long-winded methods!

owen-yap committed:
Add Javadoc comments for Ui class

owen-yap committed:
Add find by query feature

owen-yap committed:
Fix merge conflict

owen-yap committed:
Fix merge conflicts

owen-yap committed:
Change readme

owen-yap committed:
Fix checkstyle violations

owen-yap committed:
Improve code quality

owen-yap committed:
Add assertions

owen-yap committed:
Merge pull request #2 from owen-yap/branch-A-Assertions

Add assertions

owen-yap committed:
Revert "Add assertions"

owen-yap committed:
Merge pull request #3 from owen-yap/revert-2-branch-A-Assertions

Revert "Add assertions"

owen-yap committed:
Add launcher class

owen-yap committed:
Delete error files

owen-yap committed:
Add GUI using JavaFX and FXML

Our current text-based UI is plain and has no personality.

Adding a simple GUI using JavaFX can greatly increase the user
experience and add some personality to the text-bot.

JavaFX is used because of its cross-platform capabilities and ease of
implementing.

Refer to this tutorial to show how JavaFX was implemented:
https://se-education.org/guides/tutorials/javaFxPart1.html

owen-yap committed:
Refactor code for GUI

The current parser implementation and UI is for a text-based UI.

After switching to a GUI, a code refactoring is necessary to print the
correct inputs to the GUI.

Parser methods will be changed to return string values instead of
console prints.

This is so that String values can be printed to the GUI instead of
console print statements.

owen-yap committed:
Commit for BCD extension done previously

owen-yap committed:
Update Readme to the User Guide

owen-yap committed:
Fix typo

owen-yap committed:
Fix typo

owen-yap committed:
Fix grammer mistake

owen-yap committed:
Remove some links

owen-yap committed:
Add link to SE-edu addressbook

owen-yap committed:
Add greet and exit functionality to gui

owen-yap committed:
Fix some checkstyle errors
